downloads a small subset of files from DR1 (iron)

forero opened the pull request:
I am adding a script that

* finds targets near a specific point in the sky in the zall-pix iron catalog.
* finds which helpix has the most targets around that location and which tile has the larges overlap with that healpix.
* It downloads the corresponding healpix directory and the tile directory.

By default, the script looks around the COSMOS location RA=150.12, Dec=2.21 within a radius of 0.5dec, but users can change these coordinates using command-line arguments.

sbailey commented on bin/desi_download_dr1:
```diff
@@ -0,0 +1,294 @@
+    print(f"Files will be saved to {local_base_path}")
+    print(f"\nSearching for targets around RA={args.ra}, Dec={args.dec} with radius={args.radius} degrees")
+    
+    # Download redshift catalog first
```
For the purposes of making a tutorial subset, let's also download the `{specprod}/tiles-{specprod}.csv` and `{specprod}/exposures-{specprod}.csv` files (or the .fits versions).

forero replied:
Done

forero commented on bin/desi_download_dr1:
```diff
@@ -0,0 +1,294 @@
+import os
+import fsspec
+import aiohttp
+import urllib3
```
Done. Removed fsspec and aiohttp.

forero commented on bin/desi_download_dr1:
```diff
@@ -0,0 +1,294 @@
+import shutil
+import numpy as np
+import fitsio
+from astropy.table import Table
```
it is required to do it fast enough.

forero commented on bin/desi_download_dr1:
```diff
@@ -0,0 +1,294 @@
+#!/usr/bin/env python
+
+"""
+Script to download DESI DR1 data, finding healpix with most targets near a position.
```
I renamed the file before the PR review to desi_download_tiny_dr1. Updated the docstring.

forero replied:
now its `desi_get_dr_subset`

forero commented on bin/desi_download_dr1:
```diff
@@ -0,0 +1,294 @@
+    try:
+        with fsspec.open(url, 'rb', **options) as src:
+            with open(local_path, 'wb') as dst:
+                shutil.copyfileobj(src, dst)
```
Done.

forero commented on bin/desi_download_dr1:
```diff
@@ -0,0 +1,294 @@
+    parser.add_argument('--base-dir', default='./tiny_dr1', help='Base directory for downloads')
+    args = parser.parse_args()
+    
+    remote_base_url = 'https://data.desi.lbl.gov/public/dr1/'
```
tried this with the edr, but it gets complicated because the redshift catalog doesn't have the same structure (it doesn't use v1/). So I am not including edr, only dr1 and dr2

forero replied:
I also included other specprods

forero commented on bin/desi_download_dr1:
```diff
@@ -0,0 +1,294 @@
+    parser = argparse.ArgumentParser(description='Download DESI DR1 data for targets near a position')
+    parser.add_argument('--ra', type=float, default=150.11916667, help='Center right ascension in degrees (default: 150.11916667)')
+    parser.add_argument('--dec', type=float, default=2.20583333, help='Center declination in degrees (default: 2.20583333)')
+    parser.add_argument('--radius', type=float, default=0.5, help='Search radius in degrees (default: 0.5)')
```
Done. That's the center of the COSMOS field.

forero commented on bin/desi_download_dr1:
```diff
@@ -0,0 +1,294 @@
+    parser.add_argument('--ra', type=float, default=150.11916667, help='Center right ascension in degrees (default: 150.11916667)')
+    parser.add_argument('--dec', type=float, default=2.20583333, help='Center declination in degrees (default: 2.20583333)')
+    parser.add_argument('--radius', type=float, default=0.5, help='Search radius in degrees (default: 0.5)')
+    parser.add_argument('--base-dir', default='./tiny_dr1', help='Base directory for downloads')
```
I don't see the point in setting a DESI wide location as a default.

forero commented:
@sbailey I think this is ready to review. I reverted to the old default location for DR1 to be consistent with the tutorials. I also added the --no-tiles option. I also added support to download EDR data (default location is rosette 1 for dark targets).

forero commented:
Using the default settings, tiny_edr, tiny_dr1 and tiny_dr2 weigh 102GB, 40GB and 91GB, respectively (using the --no-tiles option they only weigh 9.2GB, 23GB and 51GB)
